### PR TITLE
chore: methods mocks

### PR DESCRIPTION
**Pull request by BeroBurny**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,16 +1,62 @@
 import { Web3PluginBase } from "web3";
+import { StarknetRpcApi } from "./types";
+import * as _methods from "./methods";
 
-export class StarknetPlugin extends Web3PluginBase {
+export class StarknetPlugin extends Web3PluginBase<StarknetRpcApi> {
   public pluginNamespace = "starknet";
 
   public test(param: string): void {
```
Comment (mpetrunic) [inline src/index.ts:8]: remove^^

File: src/types.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { StarknetRPCMethods } from "./enums";
+
+export type StarknetRpcApi = {
+  [StarknetRPCMethods.call]: () => Object;
```
Comment (mpetrunic) [inline src/types.ts:4]: I assume you will chaneg response type in future PR?

**Review by mpetrunic — COMMENTED** [review]
Why chore? Shouldn't this be feat?

File: src/index.ts (hunk 1 above)
Comment (BeroBurny) [reply src/index.ts:8]: It should stay as `requestManager` extends that type (via `Web3Context` that uses this generic)

Comment (BeroBurny) [reply src/index.ts:8]: oh yeah and it defaults to `type Web3APISpec = Record<string, (...params: any) => any> | unknown;` if generic is not specified

Comment (mpetrunic) [reply src/index.ts:8]: test function?!

Comment (BeroBurny) [reply src/index.ts:8]: will be removed

**Review by mpetrunic — APPROVED** [review]

**Pull request merged**